instrumentation/spring/spring-boot-autoconfigure/src/test/java/io/opentelemetry/instrumentation/spring/autoconfigure/resources/SpringBootServiceNameGuesserTest.java

```diff
@@ -0,0 +1,141 @@
+    Resource result = guesser.createResource(config);
+    expectServiceName(result, "dog-store");
+  }
+
+  @Test
+  void propertiesFileInCurrentDir() throws Exception {
+    Path propsPath = Paths.get(PROPS);
+    try {
+      writeString(propsPath, "spring.application.name=fish-tank\n");
+      when(system.openFile(PROPS)).thenCallRealMethod();
+      SpringBootServiceNameGuesser guesser = new SpringBootServiceNameGuesser(system);
+      Resource result = guesser.createResource(config);
+      expectServiceName(result, "fish-tank");
+    } finally {
+      Files.delete(propsPath);
+    }
+  }
+
+  @Test
+  void classpathApplicationYaml() {
+    when(system.openClasspathResource(APPLICATION_YML)).thenCallRealMethod();
+    SpringBootServiceNameGuesser guesser = new SpringBootServiceNameGuesser(system);
+    Resource result = guesser.createResource(config);
+    expectServiceName(result, "cat-store");
+  }
+
+  @Test
+  void yamlFileInCurrentDir() throws Exception {
+    Path yamlPath = Paths.get(APPLICATION_YML);
+    try {
+      URL url = getClass().getClassLoader().getResource(APPLICATION_YML);
+      String content = readString(Paths.get(url.toURI()));
+      writeString(yamlPath, content);
+      when(system.openFile(PROPS)).thenThrow(new FileNotFoundException());
+      when(system.openFile(APPLICATION_YML)).thenCallRealMethod();
+      SpringBootServiceNameGuesser guesser = new SpringBootServiceNameGuesser(system);
+      Resource result = guesser.createResource(config);
+      expectServiceName(result, "cat-store");
+    } finally {
+      Files.delete(yamlPath);
+    }
+  }
+
+  @Test
+  void getFromCommandlineArgsWithProcessHandle() throws Exception {
+    when(system.attemptGetCommandLineArgsViaReflection())
+        .thenReturn(
+            new String[] {
+              "/bin/java",
+              "sweet-spring.jar",
+              "--spring.application.name=tiger-town",
+              "--quiet=never"
+            });
+    SpringBootServiceNameGuesser guesser = new SpringBootServiceNameGuesser(system);
+    Resource result = guesser.createResource(config);
+    expectServiceName(result, "tiger-town");
+  }
+
+  @Test
+  void getFromCommandlineArgsWithSystemProperty() throws Exception {
+    when(system.getProperty("sun.java.command"))
+        .thenReturn("/bin/java sweet-spring.jar --spring.application.name=bullpen --quiet=never");
+    SpringBootServiceNameGuesser guesser = new SpringBootServiceNameGuesser(system);
+    Resource result = guesser.createResource(config);
+    expectServiceName(result, "bullpen");
+  }
+
+  private static void expectServiceName(Resource result, String expected) {
+    assertThat(result.getAttribute(SERVICE_NAME)).isEqualTo(expected);
+  }
+
+  private static void writeString(Path path, String value) throws Exception {
+    try (OutputStream out = Files.newOutputStream(path)) {
+      out.write(value.getBytes(UTF_8));
+    }
+  }
+
+  private static String readString(Path path) throws Exception {
```

Comment:
Hmm, how about using `Files.readAllBytes()` or `Files.readAllLines()` instead? These should be available in Java 8